tasks/Qc.wdl

```diff
@@ -282,27 +282,30 @@ task CollectDuplicateMetrics {
 task BuildBamIndex {
   input {
     File input_bam
-    File ref_dict
-    File ref_fasta
-    File ref_fasta_index
+    File ref_cache
     Int preemptible_tries
   }
 
-  String bam_index_file_name = sub(input_bam,"am$","ai")
-  Float ref_size = size(ref_fasta, "GiB") + size(ref_fasta_index, "GiB") + size(ref_dict, "GiB")
+  Float ref_size = size(ref_cache, "GiB") * 5 + 1.0)
   Int disk_size = ceil(size(input_bam, "GiB") + ref_size) + 20
 
   command {
-    /usr/local/bin/samtools index ~{input_bam} ~{bam_index_file_name}
+    # build the reference sequence cache
+    tar -zxf ~{ref_cache}
+    export REF_PATH=./cache/%2s/%2s/%s
```

Comment:
Are these URL encoded or something? Or is this a convention I'm just ignorant of.

Reply:
From Samtools:
```
In this example “%2s/%2s/%s” means the first two digits of the M5 field followed by slash, the next two digits and slash, and then the remaining 28 digits. This helps to avoid one large directory with thousands of files in it.
```

http://www.htslib.org/workflow/#mapping_to_cram